google-cloud-spanner/src/main/java/com/google/cloud/spanner/Value.java

```diff
@@ -450,6 +536,85 @@ public static Value pgJsonbArray(@Nullable Iterable<String> v) {
     return new PgJsonbArrayImpl(v == null, v == null ? null : immutableCopyOf(v));
   }
 
+  /**
+   * Returns an {@code ARRAY<PROTO>} value.
+   *
+   * @param v the source of element values. This may be {@code null} to produce a value for which
+   *     {@code isNull()} is {@code true}. Individual elements may also be {@code null}.
+   * @param descriptor Proto Type Descriptor, use static method from proto class {@code
+   *     MyProtoClass.getDescriptor()}.
+   */
+  public static Value protoMessageArray(
+      @Nullable Iterable<AbstractMessage> v, Descriptor descriptor) {
+    if (v == null) {
+      return new ProtoMessageArrayImpl(true, null, descriptor.getFullName());
+    }
+
+    List<ByteArray> serializedArray = new ArrayList<>();
+    v.forEach(
+        (message) -> {
+          if (message != null) {
+            serializedArray.add(ByteArray.copyFrom(message.toByteArray()));
```

Comment:
nit: It would have been great if we could have done this conversion without copying the byte array twice (`message.toByteArray()` creates a `new byte[]`, and the `copyFrom` method then makes a copy of that again).

I had a quick look, and could not find anything, so feel free to ignore.